src/parser/parse_tree_cl.c

```diff
@@ -2200,71 +2200,71 @@ parser_new_node (PARSER_CONTEXT * parser, PT_NODE_TYPE node_type)
   node = parser_create_node (parser);
   if (node)
     {
-      node->node_type = node_type;
+      parser_init_node (node, node_type);
       pt_parser_line_col (node);
-      parser_init_node (node);
-
       node->sql_user_text = g_query_string;
       node->sql_user_text_len = g_query_string_len;
     }
   return node;
 }
 
 /*
- * parser_init_node() - initialize a node
+ * parser_init_node() - initialize a node (Used when initializing the node for the first time after creation)
+ *   return:
+ *   node(in/out):
+ */
+PT_NODE *
+parser_init_node (PT_NODE * node, PT_NODE_TYPE node_type)
+{
+  assert (node_type < PT_LAST_NODE_NUMBER);
+  int parser_id = node->parser_id;
+
```

Comment:
Null checking for `node` is needed same like `parser_reinit_node ()`

Reply:
suggest assert(node) or if (node) as like prior code
```suggestion
PT_NODE *
parser_init_node (PT_NODE * node, PT_NODE_TYPE node_type)
{
  assert (node);
     or
  if (node)
  {
     assert (node_type < PT_LAST_NODE_NUMBER);
     int parser_id = node->parser_id;
```

Reply:
In current codes, unlike parser_reinit_node(), parser_init_node() does not have the possibility that Node will be passed as null.

Reply:
How about adding 'assert (node != NULL)' as a defense code?

Reply:
Yes, let's add assert().